pkg/controller/packageinfo/controller.go

```diff
@@ -0,0 +1,127 @@
+	"github.com/aquaproj/aqua/v2/pkg/config/aqua"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	errMissingPackages = errors.New("could not find all packages")
+)
+
+type Controller struct {
+	stdout            io.Writer
+	configFinder      ConfigFinder
+	configReader      ConfigReader
+	registryInstaller RegistryInstaller
+	fs                afero.Fs
+}
+
+type ConfigReader interface {
+	Read(logE *logrus.Entry, configFilePath string, cfg *aqua.Config) error
+}
+
+type RegistryInstaller interface {
+	InstallRegistries(ctx context.Context, logE *logrus.Entry, cfg *aqua.Config, cfgFilePath string, checksums *checksum.Checksums) (map[string]*registry.Config, error)
+}
+
+type ConfigFinder interface {
+	Find(wd, configFilePath string, globalConfigFilePaths ...string) (string, error)
+	Finds(wd, configFilePath string) []string
+}
+
+func New(configFinder ConfigFinder, configReader ConfigReader, registInstaller RegistryInstaller, fs afero.Fs) *Controller {
+	return &Controller{
+		stdout:            os.Stdout,
+		configFinder:      configFinder,
+		configReader:      configReader,
+		registryInstaller: registInstaller,
+		fs:                fs,
+	}
+}
+
+func (c *Controller) PackageInfo(ctx context.Context, param *config.Param, logE *logrus.Entry, args ...string) error {
+	cfgFilePath, err := c.configFinder.Find(param.PWD, param.ConfigFilePath, param.GlobalConfigFilePaths...)
+	if err != nil {
+		return err
+	}
+
+	cfg := &aqua.Config{}
+	if err := c.configReader.Read(logE, cfgFilePath, cfg); err != nil {
+		return err
+	}
+
+	var checksums *checksum.Checksums
+	if cfg.ChecksumEnabled(param.EnforceChecksum, param.Checksum) {
+		checksums = checksum.New()
+		checksumFilePath, err := checksum.GetChecksumFilePathFromConfigFilePath(c.fs, cfgFilePath)
+		if err != nil {
+			return err
+		}
+		if err := checksums.ReadFile(c.fs, checksumFilePath); err != nil {
+			return fmt.Errorf("read a checksum JSON: %w", err)
+		}
+		defer func() {
+			if err := checksums.UpdateFile(c.fs, checksumFilePath); err != nil {
+				logE.WithError(err).Error("update a checksum file")
+			}
+		}()
+	}
+
+	registryContents, err := c.registryInstaller.InstallRegistries(ctx, logE, cfg, cfgFilePath, checksums)
+	if err != nil {
+		return err
+	}
+
+	packageInfoByNameByRegistry := make(map[string]map[string]*registry.PackageInfo)
+	for registryName, registryConfig := range registryContents {
+		packageInfoByNameByRegistry[registryName]= registryConfig.PackageInfos.ToMap(logE)
+	}
+
+	encoder := json.NewEncoder(os.Stdout)
+	encoder.SetIndent("", "  ")
+
+	missingPackages := false
+	for _, name := range args {
+		registryName, pkgName := registryAndPackage(name)
+		pkgInfo, ok := packageInfoByNameByRegistry[registryName][pkgName]
+		if !ok {
+			logE.WithFields(logrus.Fields {
+				"registry_name": registryName,
+				"package_name": pkgName,
+			}).Error("no such registry or package")
+			missingPackages = true
+			continue
+		}
+
+		if err := encoder.Encode(pkgInfo); err != nil {
```

Comment:
Hmm. This command accepts multiple package names and outputs them as JSON independently.
I think it's weird because the output can't be parsed as JSON.

```json
{
  // ...
}
{
  // ...
}
```

I think there are several options.

1. Accept only one package
2. Return a list of packages

```json
[
  {
    // ...
  },
  {
    // ...
  }
]
```

Reply:
JSON is a self-delimiting format. There is no parsing ambiguity in a stream of messages. Tools like `jq` and libraries such as Go's `encoding/json` have no issues working with such streams.

For example, with the current implementation, I'm able to aggregate a count of installed packages by owner using `jq`:
```
❮ ./aqua info package $(./aqua list --installed | awk '{print $3 "," $1}') | jq -r '.repo_owner' | sort | uniq -c 
   1 golangci
   1 goreleaser
   1 int128
   1 reviewdog
   1 rhysd
   1 sigstore
   2 suzuki-shunsuke
```

Of the alternatives you suggest, I think I prefer "accept only one package". Its got what seems to me like an unnecessary restriction, but the output is direct and simple to work with. It is a subset of the stream-of-objects output format I proposed: just pass a single name if you want a single message of output. If we do choose to restrict the caller to requesting only one package, what should happen if they pass more than one in `argv`?

I'd really prefer to avoid the list format. I find it over-encapsulated, less human-readable, and a bit less convenient to work with. There's not really an upside.